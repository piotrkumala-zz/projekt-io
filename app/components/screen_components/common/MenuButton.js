import React from "react";
import { View, TouchableOpacity, StyleSheet, Image, Text } from "react-native";

const MenuButton = (props) => {
  // props.handler
  // props.description
  //
  const description = props.description;
  const handler = props.handler;

  return (
    <View style={styles.view}>
      <TouchableOpacity onPress={props.handler} style={styles.button}>
        <Text style={styles.text}>{description}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  view: {
    alignSelf: "stretch",
    margin: 10,
  },
  button: {
    backgroundColor: "#ddd",
    alignItems: "center",
    borderRadius: 50,
    padding: 15,
  },
  text: {
    fontSize: 24,
    textAlign: "center",
    alignItems: "center",
    color: "#777",
  },
});

export default MenuButton;
