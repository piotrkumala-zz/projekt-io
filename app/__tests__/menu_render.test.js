import React from 'react';
import MenuButton from '../components/screen_components/common/MenuButton';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<MenuButton />).toJSON();
  expect(tree).toMatchSnapshot();
});