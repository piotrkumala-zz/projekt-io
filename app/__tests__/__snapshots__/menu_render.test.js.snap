// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "margin": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ddd",
        "borderRadius": 50,
        "opacity": 1,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "color": "#777",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
